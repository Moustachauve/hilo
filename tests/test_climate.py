"""Tests for the Hilo switch platform."""

from unittest.mock import MagicMock

import pytest
from homeassistant.const import (
    Platform,
)
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er
from pytest_homeassistant_custom_component.common import MockConfigEntry
from syrupy.assertion import SnapshotAssertion

from . import setup_with_selected_platforms


@pytest.mark.usefixtures("entity_registry_enabled_by_default", "mock_api")
async def test_climate(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    mock_api: MagicMock,
) -> None:
    """Test the creation and values of the Hilo Climate."""
    await setup_with_selected_platforms(
        hass, mock_config_entry, [Platform.CLIMATE], mock_api
    )

    entity_entries = er.async_entries_for_config_entry(
        entity_registry, mock_config_entry.entry_id
    )

    assert entity_entries
    for entity_entry in entity_entries:
        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
        assert (state := hass.states.get(entity_entry.entity_id))
        assert state == snapshot(name=f"{entity_entry.entity_id}-state")
