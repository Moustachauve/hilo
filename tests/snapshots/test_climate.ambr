# serializer version: 1
# name: test_climate[climate.thermostat_1-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'hvac_modes': list([
        <HVACMode.HEAT: 'heat'>,
      ]),
      'max_temp': 36.0,
      'min_temp': 5.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'climate',
    'entity_category': None,
    'entity_id': 'climate.thermostat_1',
    'has_entity_name': False,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:radiator-disabled',
    'original_name': 'Thermostat 1',
    'platform': 'hilo',
    'previous_unique_id': None,
    'supported_features': <ClimateEntityFeature: 1>,
    'translation_key': None,
    'unique_id': '1a2b3c4d5e6f-climate',
    'unit_of_measurement': None,
  })
# ---
# name: test_climate[climate.thermostat_1-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Thermostat 1',
      'hvac_modes': list([
        <HVACMode.HEAT: 'heat'>,
      ]),
      'icon': 'mdi:radiator-disabled',
      'max_temp': 36.0,
      'min_temp': 5.0,
      'supported_features': <ClimateEntityFeature: 1>,
    }),
    'context': <ANY>,
    'entity_id': 'climate.thermostat_1',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
